### PR TITLE
Homework Tools #2

### PR DESCRIPTION
**Pull request by jakjan95**
Solution to homework from tools lesson

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,3 +1,3 @@
 all: src/*.cpp inc/*.hpp
-	g++ -std=c++17 src/*.cpp -o main
+	g++ -std=c++17 -Werror -Iinc/  src/*.cpp -o main
```
Comment (Kasperice) [inline Makefile]: you can consider using more flags: -Wall -Wextra -pedantic

File: src/functions.cpp [hunk 2]
```diff
@@ -1,6 +1,7 @@
 #include "../inc/functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
```
Comment (Kasperice) [inline src/functions.cpp]: It should be also changed to `#include "functions.hpp"`

File: src/functions.cpp [hunk 3]
```diff
@@ -1,6 +1,7 @@
...
 #include <iostream>
 #include <limits>
+#include <algorithm>
```
Comment (Marcinati) [inline src/functions.cpp]: You have proper flag (-Iinc/), so as Kasperice says, you can add headers straight by name.
Second case: sort headers. ;)

**Review by Marcinati — CHANGES_REQUESTED** [review]

File: Makefile [hunk 4]
```diff
@@ -1,3 +1,3 @@
 all: src/*.cpp inc/*.hpp
-	g++ -std=c++17 src/*.cpp -o main
+	g++ -std=c++17 -Werror  src/*.cpp -o main
```
Comment (ziobron) [inline Makefile]: Not only `-Werror` should be added. Please add `-Wall -Wextra -pedantic -Wconversion`

**Review by ziobron — CHANGES_REQUESTED** [review]

File: src/functions.cpp [hunk 5]
```diff
@@ -1,6 +1,7 @@
...
 
 void print(std::vector<int> const & numbers) {
     for (int i = 0; i < numbers.size(); ++i) {
```
Comment (DamiXioR) [inline src/functions.cpp]: you should use size_t

File: Makefile (hunk 1 above)
Comment (DamiXioR) [inline Makefile]: -I. /inc looks better

**Review by DamiXioR — COMMENTED** [review]
nice work

**Comment by ziobron** [comment]
Great!

+4 XP Narzędzia#2 – vimtutor
+5 XP Narzędzia#2 – git - konflikty
+4 XP Narzędzia#2 – flagi kompilacji
+3 XP Narzędzia#2 – git - cherrypick
+8 XP delivery before 07.06.2020

{ "requirement_ids": [1119, 1999, 1119, 2000, 1119, 2001, 1119, 2002], "points": [2, 4, 2, 5, 2, 4, 2, 3] }

**Pull request closed without merging**